consensus/safety-rules/src/lib.rs

```diff
@@ -186,19 +150,51 @@ impl SafetyRules {
         &self,
         block_parent_qc: &QuorumCert,
         block_round: u64,
-    ) -> Option<HashValue> {
+    ) -> bool {
```

Comment:
What about returning `Option<BlockInfo>` for the committed block? It'd keep the choice of the potential commit in one place. 
The caller would then `if let Some(bi) = self. commit_rule_for_certified_block...`
I don't think it'd complicate the testing too much. At the very least you can convert all the tests from true / false to is_some() / is_none() :)

Reply:
I'd prefer we not use options as we already know we'll either have a empty ledger info or the parent in the qc. How about I just merge the commit rule test into the code and update the tests to check the id matches the expected id (zero or the actual)?

Reply:
Let me know what you think of the new version, admittedly I may be relying too much on my non-rust experience... so if it still feels funny, I'll go with your approach.